src/util/projection.ts

```diff
@@ -0,0 +1,28 @@
+export default class ProjectionDefinition {
+    constructor(from: string, to: string, transition: number){
+        this[0] = from;
```

Comment:
This is strange for typescript, why not use "this.from" etc?

Reply:
it'll serialize as:
{from:this.from , to:this.to , t:this.t}

The current constructor will serialize as array
[from, to, t]

Reply:
Serialization is defined in `toString` isn't it? I prefer to know which fields a class has when reading it, this array like thing is confusing, to me...

Reply:
It only became "confusing Array like thing" when I removed the `extends Array` per your request. Before that it was an actual Array.

Reply:
I still think it should be a regular class, I don't understand the value of having it look like an array internally, can you explain the added value of this?
```ts
export default class ProjectionDefinition {
  constructor(private from: string, private to: string, private transition: number) {}

  toString() {
    return ...
  }
  // maybe getter here, or simply make the properties public
  ...
  
}
```

Reply:
If we can make it serialized correctly then it's fine with me to have the from / to / t

Reply:
Where is this serialization happening? What needs to be added in order to make it work like you would like it to work?

Reply:
It's here

https://github.com/maplibre/maplibre-style-spec/blob/4ebf7975507637282b59b37a156cb024578f15ea/test/integration/expression/expression.test.ts#L60